headers.h

```diff
@@ -11,15 +11,30 @@
 #define HAVE_SETSID
 
 /* OS-specific includes */
-#if defined(__GLIBC__)
+#if defined(__GLIBC__) || defined(__CYGWIN__)
+# define HAVE_UNISTD_H
+# define HAVE_SYS_IOCTL_H
 # define HAVE_SYS_SYSMACROS_H
 # define HAVE_ERROR_H
```

Comment:
In a tangential direction: It seems that the `<error.h>` header isn't used at all.